src/Sylius/Behat/Client/ApiPlatformClient.php

```diff
@@ -38,214 +40,100 @@ public function __construct(AbstractBrowser $client, SharedStorageInterface $sha
-    }
-
-    public function subResourceIndex(string $resource, string $subResource, string $id): void
-    {
-        $this->request('GET', sprintf('/new-api/%s/%s/%s', $resource, $id, $subResource), ['HTTP_ACCEPT' => 'application/ld+json']);
-    }
-
-    public function buildCreateRequest(string $resource): void
-    {
-        $this->request['url'] = '/new-api/' . $resource;
-    }
-
-    public function buildUpdateRequest(string $resource, string $id): void
-    {
-        $this->show($resource, $id);
-
-        $this->request['url'] = sprintf('/new-api/%s/%s', $resource, $id);
-        $this->request['body'] = json_decode($this->client->getResponse()->getContent(), true);
-    }
-
-    public function buildFilter(array $filters): void
-    {
-        $this->filters = $filters;
+        $this->request(Request::custom($iri, 'GET', $this->sharedStorage->get('token')));
     }
 
-    /** @param string|int $value */
-    public function addRequestData(string $key, $value): void
+    public function subResourceIndex(string $subResource, string $id): void
     {
-        $this->request['body'][$key] = $value;
+        $this->request(Request::subResourceIndex($this->resource, $id, $subResource, $this->sharedStorage->get('token')));
     }
 
-    public function addCompoundRequestData(array $data): void
+    public function show(string $id): void
     {
-        $this->request['body'] = array_merge_recursive($this->request['body'], $data);
-    }
-
-    public function updateRequestData(array $data): void
-    {
-        $this->request['body'] = $this->mergeArraysUniquely($this->request['body'], $data);
+        $this->request(Request::show($this->resource, $id, $this->sharedStorage->get('token')));
     }
 
     public function create(): void
     {
-        $content = json_encode($this->request['body']);
-
-        $this->request('POST', $this->request['url'], ['CONTENT_TYPE' => 'application/json'], $content);
+        $this->request($this->request);
     }
 
     public function update(): void
     {
-        $content = json_encode($this->request['body']);
-
-        $this->request('PUT', $this->request['url'], ['CONTENT_TYPE' => 'application/ld+json'], $content);
+        $this->request($this->request);
     }
 
-    public function delete(string $resource, string $id): void
+    public function delete(string $id): void
     {
-        $this->request('DELETE', sprintf('/new-api/%s/%s', $resource, $id), []);
+        $this->request(Request::delete($this->resource, $id, $this->sharedStorage->get('token')));
     }
 
-    public function filter(string $resource): void
+    public function filter(): void
     {
         $query = http_build_query($this->filters, '', '&', PHP_QUERY_RFC3986);
-        $path = sprintf('/new-api/%s?%s', $resource, $query);
-
-        $this->request('GET', $path, ['HTTP_ACCEPT' => 'application/ld+json']);
-    }
+        $path = sprintf('/new-api/%s?%s', $this->resource, $query);
```

Comment:
Could we include a dedicated method in `Request` object?

Reply:
Yes, I've mentioned it in the PR description 👍